app/connector/aws-ddb/src/test/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataAdapterTest.java

```diff
@@ -0,0 +1,82 @@
+ * limitations under the License.
+ */
+package io.syndesis.connector.aws.ddb.metadata;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.syndesis.common.util.Json;
+import io.syndesis.connector.support.verifier.api.SyndesisMetadata;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.MetaDataExtension.MetaData;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.junit.Test;
+
+public class AWSDDBMetadataAdapterTest {
+
+
+    @Test
+    public void adaptTest() throws JsonProcessingException {
+        CamelContext camelContext = new DefaultCamelContext();
+        AWSDDBConnectorMetaDataExtension ext = new AWSDDBConnectorMetaDataExtension(camelContext);
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("element", "{\"clave\" : \":#KEY\"}");
+        parameters.put("attributes", "clave, atributo");
+        parameters.put("region", "EU_WEST_3");
+        parameters.put("secretKey", "invalidKey");
+        parameters.put("accessKey", "invalidKey");
+        parameters.put("tableName", "TestTable");
+        Optional<MetaData> metadata = ext.meta(parameters);
+        AWSDDBMetadataRetrieval adapter = new AWSDDBMetadataRetrieval();
+
+        ObjectWriter writer = Json.writer();
+        SyndesisMetadata syndesisMetaData2 = adapter.adapt(camelContext,
+        "aws-ddb", "io.syndesis:aws-ddb-query-connector", parameters, metadata.get());
+        String actualMetadata = writer.with(writer.getConfig().getDefaultPrettyPrinter()).writeValueAsString(syndesisMetaData2);
+
+        String expectedMetadata = "{\n" +
+        "  \"inputShape\" : {\n" +
+        "    \"name\" : \"Parameters\",\n" +
+        "    \"description\" : \"Query parameters.\",\n" +
+        "    \"kind\" : \"json-schema\",\n" +
+        "    \"type\" : \"Parameters\",\n" +
+        "    \"specification\" : \"{\\\"type\\\":\\\"object\\\",\\\"title\\\":\\\"Parameters\\\",\\\"properties\\\":{\\\":#KEY\\\":{\\\"type\\\":\\\"string\\\"}}}\",\n" +
+        "    \"metadata\" : {\n" +
+        "      \"variant\" : \"element\"\n" +
+        "    }\n" +
+        "  },\n" +
+        "  \"outputShape\" : {\n" +
+        "    \"name\" : \"Result\",\n" +
+        "    \"description\" : \"Attributes on the result.\",\n" +
+        "    \"kind\" : \"json-schema\",\n" +
+        "    \"type\" : \"Result\",\n" +
+        "    \"specification\" : \"{\\\"type\\\":\\\"object\\\",\\\"title\\\":\\\"Result\\\",\\\"properties\\\":{\\\"clave\\\":{\\\"type\\\":\\\"string\\\"},\\\"atributo\\\":{\\\"type\\\":\\\"string\\\"}}}\",\n" +
+        "    \"metadata\" : {\n" +
+        "      \"variant\" : \"collection\"\n" +
+        "    }\n" +
+        "  }\n" +
+        "}";
+
+        assertEquals(expectedMetadata, actualMetadata);
```

Comment:
Perhaps we could use [JSONAssert](http://jsonassert.skyscreamer.org/) here. Example of the use can be found here:

https://github.com/syndesisio/syndesis/blob/4b628bcbab6fef93684e741a6fae879c729ae31a/app/common/util/src/test/java/io/syndesis/common/util/openapi/OpenApiHelperTest.java#L50-L59

Reply:
:+1: